tools/validate/validate.cpp

```diff
@@ -0,0 +1,205 @@
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#define _CRT_SECURE_NO_WARNINGS
+#include <algorithm>
+#include <array>
+#include <assert.h>
+#include <filesystem>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string>
+#include <string_view>
+#include <utility>
+#include <vector>
+using namespace std;
+
+class BinaryFile {
+public:
+    explicit BinaryFile(const filesystem::path& filepath) {
+        m_file = _wfopen(filepath.c_str(), L"rb");

```

Comment:
our filesystem::path is <wchar_t> correct?

Reply:
That is correct.